File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/InterestLeg.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class InterestLeg : DirectionalLeg
    {
        public InterestLeg(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode interestLegCalculationPeriodDatesNode = xmlNode.SelectSingleNode("interestLegCalculationPeriodDates");
            
            if (interestLegCalculationPeriodDatesNode != null)
            {
                if (interestLegCalculationPeriodDatesNode.Attributes["href"] != null || interestLegCalculationPeriodDatesNode.Attributes["id"] != null) 
                {
                    if (interestLegCalculationPeriodDatesNode.Attributes["id"] != null) 
                    {
                        interestLegCalculationPeriodDatesIDRef_ = interestLegCalculationPeriodDatesNode.Attributes["id"].Value;
                        InterestLegCalculationPeriodDates ob = new InterestLegCalculationPeriodDates(interestLegCalculationPeriodDatesNode);
                        IDManager.SetID(interestLegCalculationPeriodDatesIDRef_, ob);
                    }
                    else if (interestLegCalculationPeriodDatesNode.Attributes["href"] != null)
                    {
                        interestLegCalculationPeriodDatesIDRef_ = interestLegCalculationPeriodDatesNode.Attributes["href"].Value;
                    }
                    else
                    {
                        interestLegCalculationPeriodDates_ = new InterestLegCalculationPeriodDates(interestLegCalculationPeriodDatesNode);
                    }
                }
                else
                {
                    interestLegCalculationPeriodDates_ = new InterestLegCalculationPeriodDates(interestLegCalculationPeriodDatesNode);
                }
            }
            
        
            XmlNode notionalNode = xmlNode.SelectSingleNode("notional");
            
            if (notionalNode != null)
            {
                if (notionalNode.Attributes["href"] != null || notionalNode.Attributes["id"] != null) 
                {
                    if (notionalNode.Attributes["id"] != null) 
                    {
                        notionalIDRef_ = notionalNode.Attributes["id"].Value;
                        ReturnSwapNotional ob = new ReturnSwapNotional(notionalNode);
                        IDManager.SetID(notionalIDRef_, ob);
                    }
                    else if (notionalNode.Attributes["href"] != null)
                    {
                        notionalIDRef_ = notionalNode.Attributes["href"].Value;
                    }
                    else
                    {
                        notional_ = new ReturnSwapNotional(notionalNode);
                    }
                }
                else
                {
                    notional_ = new ReturnSwapNotional(notionalNode);
                }
            }
            
        
            XmlNode interestAmountNode = xmlNode.SelectSingleNode("interestAmount");
            
            if (interestAmountNode != null)
            {
                if (interestAmountNode.Attributes["href"] != null || interestAmountNode.Attributes["id"] != null) 
                {
                    if (interestAmountNode.Attributes["id"] != null) 
                    {
                        interestAmountIDRef_ = interestAmountNode.Attributes["id"].Value;
                        LegAmount ob = new LegAmount(interestAmountNode);
                        IDManager.SetID(interestAmountIDRef_, ob);
                    }
                    else if (interestAmountNode.Attributes["href"] != null)
                    {
                        interestAmountIDRef_ = interestAmountNode.Attributes["href"].Value;
                    }
                    else
                    {
                        interestAmount_ = new LegAmount(interestAmountNode);
                    }
                }
                else
                {
                    interestAmount_ = new LegAmount(interestAmountNode);
                }
            }
            
        
            XmlNode interestCalculationNode = xmlNode.SelectSingleNode("interestCalculation");
            
            if (interestCalculationNode != null)
            {
                if (interestCalculationNode.Attributes["href"] != null || interestCalculationNode.Attributes["id"] != null) 
                {
                    if (interestCalculationNode.Attributes["id"] != null) 
                    {
                        interestCalculationIDRef_ = interestCalculationNode.Attributes["id"].Value;
                        InterestCalculation ob = new InterestCalculation(interestCalculationNode);
                        IDManager.SetID(interestCalculationIDRef_, ob);
                    }
                    else if (interestCalculationNode.Attributes["href"] != null)
                    {
                        interestCalculationIDRef_ = interestCalculationNode.Attributes["href"].Value;
                    }
                    else
                    {
                        interestCalculation_ = new InterestCalculation(interestCalculationNode);
                    }
                }
                else
                {
                    interestCalculation_ = new InterestCalculation(interestCalculationNode);
                }
            }
            
        
            XmlNode stubCalculationPeriodNode = xmlNode.SelectSingleNode("stubCalculationPeriod");
            
            if (stubCalculationPeriodNode != null)
            {
                if (stubCalculationPeriodNode.Attributes["href"] != null || stubCalculationPeriodNode.Attributes["id"] != null) 
                {
                    if (stubCalculationPeriodNode.Attributes["id"] != null) 
                    {
                        stubCalculationPeriodIDRef_ = stubCalculationPeriodNode.Attributes["id"].Value;
                        StubCalculationPeriod ob = new StubCalculationPeriod(stubCalculationPeriodNode);
                        IDManager.SetID(stubCalculationPeriodIDRef_, ob);
                    }
                    else if (stubCalculationPeriodNode.Attributes["href"] != null)
                    {
                        stubCalculationPeriodIDRef_ = stubCalculationPeriodNode.Attributes["href"].Value;
                    }
                    else
                    {
                        stubCalculationPeriod_ = new StubCalculationPeriod(stubCalculationPeriodNode);
                    }
                }
                else
                {
                    stubCalculationPeriod_ = new StubCalculationPeriod(stubCalculationPeriodNode);
                }
            }
            
        
        }
        
    
        #region InterestLegCalculationPeriodDates_
        private InterestLegCalculationPeriodDates interestLegCalculationPeriodDates_;
        public InterestLegCalculationPeriodDates InterestLegCalculationPeriodDates_
        {
            get
            {
                if (this.interestLegCalculationPeriodDates_ != null)
                {
                    return this.interestLegCalculationPeriodDates_; 
                }
                else if (this.interestLegCalculationPeriodDatesIDRef_ != null)
                {
                    interestLegCalculationPeriodDates_ = IDManager.getID(interestLegCalculationPeriodDatesIDRef_) as InterestLegCalculationPeriodDates;
                    return this.interestLegCalculationPeriodDates_; 
                }
                else
                {
                      return this.interestLegCalculationPeriodDates_; 
                }
            }
            set
            {
                if (this.interestLegCalculationPeriodDates_ != value)
                {
                    this.interestLegCalculationPeriodDates_ = value;
                }
            }
        }
        #endregion
        
        public string interestLegCalculationPeriodDatesIDRef_ { get; set; }
        #region Notional_
        private ReturnSwapNotional notional_;
        public ReturnSwapNotional Notional_
        {
            get
            {
                if (this.notional_ != null)
                {
                    return this.notional_; 
                }
                else if (this.notionalIDRef_ != null)
                {
                    notional_ = IDManager.getID(notionalIDRef_) as ReturnSwapNotional;
                    return this.notional_; 
                }
                else
                {
                      return this.notional_; 
                }
            }
            set
            {
                if (this.notional_ != value)
                {
                    this.notional_ = value;
                }
            }
        }
        #endregion
        
        public string notionalIDRef_ { get; set; }
        #region InterestAmount_
        private LegAmount interestAmount_;
        public LegAmount InterestAmount_
        {
            get
            {
                if (this.interestAmount_ != null)
                {
                    return this.interestAmount_; 
                }
                else if (this.interestAmountIDRef_ != null)
                {
                    interestAmount_ = IDManager.getID(interestAmountIDRef_) as LegAmount;
                    return this.interestAmount_; 
                }
                else
                {
                      return this.interestAmount_; 
                }
            }
            set
            {
                if (this.interestAmount_ != value)
                {
                    this.interestAmount_ = value;
                }
            }
        }
        #endregion
        
        public string interestAmountIDRef_ { get; set; }
        #region InterestCalculation_
        private InterestCalculation interestCalculation_;
        public InterestCalculation InterestCalculation_
        {
            get
            {
                if (this.interestCalculation_ != null)
                {
                    return this.interestCalculation_; 
                }
                else if (this.interestCalculationIDRef_ != null)
                {
                    interestCalculation_ = IDManager.getID(interestCalculationIDRef_) as InterestCalculation;
                    return this.interestCalculation_; 
                }
                else
                {
                      return this.interestCalculation_; 
                }
            }
            set
            {
                if (this.interestCalculation_ != value)
                {
                    this.interestCalculation_ = value;
                }
            }
        }
        #endregion
        
        public string interestCalculationIDRef_ { get; set; }
        #region StubCalculationPeriod_
        private StubCalculationPeriod stubCalculationPeriod_;
        public StubCalculationPeriod StubCalculationPeriod_
        {
            get
            {
                if (this.stubCalculationPeriod_ != null)
                {
                    return this.stubCalculationPeriod_; 
                }
                else if (this.stubCalculationPeriodIDRef_ != null)
                {
                    stubCalculationPeriod_ = IDManager.getID(stubCalculationPeriodIDRef_) as StubCalculationPeriod;
                    return this.stubCalculationPeriod_; 
                }
                else
                {
                      return this.stubCalculationPeriod_; 
                }
            }
            set
            {
                if (this.stubCalculationPeriod_ != value)
                {
                    this.stubCalculationPeriod_ = value;
                }
            }
        }
        #endregion
        
        public string stubCalculationPeriodIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/InterestLeg.cpp
// InterestLeg.cpp 
#include "InterestLeg.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

InterestLeg::InterestLeg(TiXmlNode* xmlNode)
: DirectionalLeg(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //interestLegCalculationPeriodDatesNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* interestLegCalculationPeriodDatesNode = xmlNode->FirstChildElement("interestLegCalculationPeriodDates");

   if(interestLegCalculationPeriodDatesNode){interestLegCalculationPeriodDatesIsNull_ = false;}
   else{interestLegCalculationPeriodDatesIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- interestLegCalculationPeriodDatesNode , address : " << interestLegCalculationPeriodDatesNode << std::endl;
   #endif
   if(interestLegCalculationPeriodDatesNode)
   {
      if(interestLegCalculationPeriodDatesNode->Attribute("href") || interestLegCalculationPeriodDatesNode->Attribute("id"))
      {
          if(interestLegCalculationPeriodDatesNode->Attribute("id"))
          {
             interestLegCalculationPeriodDatesIDRef_ = interestLegCalculationPeriodDatesNode->Attribute("id");
             interestLegCalculationPeriodDates_ = boost::shared_ptr<InterestLegCalculationPeriodDates>(new InterestLegCalculationPeriodDates(interestLegCalculationPeriodDatesNode));
             interestLegCalculationPeriodDates_->setID(interestLegCalculationPeriodDatesIDRef_);
             IDManager::instance().setID(interestLegCalculationPeriodDatesIDRef_,interestLegCalculationPeriodDates_);
          }
          else if(interestLegCalculationPeriodDatesNode->Attribute("href")) { interestLegCalculationPeriodDatesIDRef_ = interestLegCalculationPeriodDatesNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { interestLegCalculationPeriodDates_ = boost::shared_ptr<InterestLegCalculationPeriodDates>(new InterestLegCalculationPeriodDates(interestLegCalculationPeriodDatesNode));}
   }

   //notionalNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* notionalNode = xmlNode->FirstChildElement("notional");

   if(notionalNode){notionalIsNull_ = false;}
   else{notionalIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- notionalNode , address : " << notionalNode << std::endl;
   #endif
   if(notionalNode)
   {
      if(notionalNode->Attribute("href") || notionalNode->Attribute("id"))
      {
          if(notionalNode->Attribute("id"))
          {
             notionalIDRef_ = notionalNode->Attribute("id");
             notional_ = boost::shared_ptr<ReturnSwapNotional>(new ReturnSwapNotional(notionalNode));
             notional_->setID(notionalIDRef_);
             IDManager::instance().setID(notionalIDRef_,notional_);
          }
          else if(notionalNode->Attribute("href")) { notionalIDRef_ = notionalNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { notional_ = boost::shared_ptr<ReturnSwapNotional>(new ReturnSwapNotional(notionalNode));}
   }

   //interestAmountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* interestAmountNode = xmlNode->FirstChildElement("interestAmount");

   if(interestAmountNode){interestAmountIsNull_ = false;}
   else{interestAmountIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- interestAmountNode , address : " << interestAmountNode << std::endl;
   #endif
   if(interestAmountNode)
   {
      if(interestAmountNode->Attribute("href") || interestAmountNode->Attribute("id"))
      {
          if(interestAmountNode->Attribute("id"))
          {
             interestAmountIDRef_ = interestAmountNode->Attribute("id");
             interestAmount_ = boost::shared_ptr<LegAmount>(new LegAmount(interestAmountNode));
             interestAmount_->setID(interestAmountIDRef_);
             IDManager::instance().setID(interestAmountIDRef_,interestAmount_);
          }
          else if(interestAmountNode->Attribute("href")) { interestAmountIDRef_ = interestAmountNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { interestAmount_ = boost::shared_ptr<LegAmount>(new LegAmount(interestAmountNode));}
   }

   //interestCalculationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* interestCalculationNode = xmlNode->FirstChildElement("interestCalculation");

   if(interestCalculationNode){interestCalculationIsNull_ = false;}
   else{interestCalculationIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- interestCalculationNode , address : " << interestCalculationNode << std::endl;
   #endif
   if(interestCalculationNode)
   {
      if(interestCalculationNode->Attribute("href") || interestCalculationNode->Attribute("id"))
      {
          if(interestCalculationNode->Attribute("id"))
          {
             interestCalculationIDRef_ = interestCalculationNode->Attribute("id");
             interestCalculation_ = boost::shared_ptr<InterestCalculation>(new InterestCalculation(interestCalculationNode));
             interestCalculation_->setID(interestCalculationIDRef_);
             IDManager::instance().setID(interestCalculationIDRef_,interestCalculation_);
          }
          else if(interestCalculationNode->Attribute("href")) { interestCalculationIDRef_ = interestCalculationNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { interestCalculation_ = boost::shared_ptr<InterestCalculation>(new InterestCalculation(interestCalculationNode));}
   }

   //stubCalculationPeriodNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* stubCalculationPeriodNode = xmlNode->FirstChildElement("stubCalculationPeriod");

   if(stubCalculationPeriodNode){stubCalculationPeriodIsNull_ = false;}
   else{stubCalculationPeriodIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- stubCalculationPeriodNode , address : " << stubCalculationPeriodNode << std::endl;
   #endif
   if(stubCalculationPeriodNode)
   {
      if(stubCalculationPeriodNode->Attribute("href") || stubCalculationPeriodNode->Attribute("id"))
      {
          if(stubCalculationPeriodNode->Attribute("id"))
          {
             stubCalculationPeriodIDRef_ = stubCalculationPeriodNode->Attribute("id");
             stubCalculationPeriod_ = boost::shared_ptr<StubCalculationPeriod>(new StubCalculationPeriod(stubCalculationPeriodNode));
             stubCalculationPeriod_->setID(stubCalculationPeriodIDRef_);
             IDManager::instance().setID(stubCalculationPeriodIDRef_,stubCalculationPeriod_);
          }
          else if(stubCalculationPeriodNode->Attribute("href")) { stubCalculationPeriodIDRef_ = stubCalculationPeriodNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { stubCalculationPeriod_ = boost::shared_ptr<StubCalculationPeriod>(new StubCalculationPeriod(stubCalculationPeriodNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<InterestLegCalculationPeriodDates> InterestLeg::getInterestLegCalculationPeriodDates()
{
   if(!this->interestLegCalculationPeriodDatesIsNull_){
        if(interestLegCalculationPeriodDatesIDRef_ != ""){
             return boost::shared_static_cast<InterestLegCalculationPeriodDates>(IDManager::instance().getID(interestLegCalculationPeriodDatesIDRef_));
        }else{
             return this->interestLegCalculationPeriodDates_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<InterestLegCalculationPeriodDates>();
   }
}
boost::shared_ptr<ReturnSwapNotional> InterestLeg::getNotional()
{
   if(!this->notionalIsNull_){
        if(notionalIDRef_ != ""){
             return boost::shared_static_cast<ReturnSwapNotional>(IDManager::instance().getID(notionalIDRef_));
        }else{
             return this->notional_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ReturnSwapNotional>();
   }
}
boost::shared_ptr<LegAmount> InterestLeg::getInterestAmount()
{
   if(!this->interestAmountIsNull_){
        if(interestAmountIDRef_ != ""){
             return boost::shared_static_cast<LegAmount>(IDManager::instance().getID(interestAmountIDRef_));
        }else{
             return this->interestAmount_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<LegAmount>();
   }
}
boost::shared_ptr<InterestCalculation> InterestLeg::getInterestCalculation()
{
   if(!this->interestCalculationIsNull_){
        if(interestCalculationIDRef_ != ""){
             return boost::shared_static_cast<InterestCalculation>(IDManager::instance().getID(interestCalculationIDRef_));
        }else{
             return this->interestCalculation_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<InterestCalculation>();
   }
}
boost::shared_ptr<StubCalculationPeriod> InterestLeg::getStubCalculationPeriod()
{
   if(!this->stubCalculationPeriodIsNull_){
        if(stubCalculationPeriodIDRef_ != ""){
             return boost::shared_static_cast<StubCalculationPeriod>(IDManager::instance().getID(stubCalculationPeriodIDRef_));
        }else{
             return this->stubCalculationPeriod_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<StubCalculationPeriod>();
   }
}
}

